internal/testutil/acc/atlas.go

```diff
@@ -0,0 +1,116 @@
+package acc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// TestMainExecution must be called from TestMain in the test package if ProjectIDExecution is going to be used.
+func TestMainExecution(m *testing.M) {
```

Comment:
any suggestion?  This method is the implementation of TestMain in the tests, it basically do some setup, execute the tests and teardown. 

I called the methods TestMainExecution and ProjectIDExecution with the intention that Execution means during the test execution of the package.     So for example ProjectIDExecution provides the id of a project that is created before the tests run and deleted after.

Reply:
looking at what this file is managing, I think this should be about project resource initialisation.

`TestMainExecution` --> `TestResourceInit`
`ProjectIDExecution` --> `GetOrCreateProjectID`

I would avoid having this generic `TestMain` logic when in reality it is solely focusing on project resource re-usage (reminds me of [YAGNI](https://martinfowler.com/bliki/Yagni.html)). 

I still have not digested this file in full (as I mentioned somewhere, it looks too complicated), looping in the rest of the team @maastha @AgustinBettati @EspenAlbert @oarbusi

Reply:
TestMainExecution is really for initialising and destroying resources, and calling test runs in the middle.

It's our custom implementation for TestMain adding resource lifecycle management

Reply:
some info here about TestMain: https://medium.com/goingogo/why-use-testmain-for-testing-in-go-dafb52b406bc

Reply:
I do agree that we should focus on more concrete naming so that it is more clear what is being done. I was more on the side of a name such as `SetupSharedProject` over `TestMainExecution`.

> TestMainExecution is really for initialising and destroying resources

Regarding this point I was slightly confused because this function does not contain all the initialising logic, for example the creation of the project is handled in `ProjectIDExecution`. Was this for a specific reason?

Reply:
I do agree that we should focus on more concrete naming so that it is more clear what is being done. I was more on the side of a name such as SetupSharedProject over TestMainExecution.

TestMainExecution is really for initialising and destroying resources

Regarding this point I was slightly confused because this function does not contain all the initialising logic, for example the creation of the project is handled in ProjectIDExecution. Was this for a specific reason?

Reply:
yes, because i realized that for instance right now we always run all packages even if one test won't be run.   so I lazy initialize the project creation in case test is being executed but doesn't need to get the execution project.

Reply:
How about `ProjectIDForExecution` or `GetProjectIDForCurrentExecution` or something?

Reply:
Additionally, I think some methods/files could use more comments as to the "why", for example shared_resource.go (or the TestMain()),  as the mechanism could be hard for someone new to understand or extend completely or in case we run into any issues later.

Reply:
I'm not sure about document in our code how Go features like TestMain work, but I'm fine to document more our code. I'm going to merge but please feel free to suggest doc, and i'll think in more docs in the following PRs, thx!